src/shared/mentionable_text_input.rs

```diff
@@ -695,32 +738,10 @@ impl MentionableTextInput {
                 return;
             }
 
-            let header_view = self.cmd_text_input.view(id!(popup.header_view));
-
-            let header_height = if header_view.area().rect(cx).size.y > 0.0 {
-                header_view.area().rect(cx).size.y
-            } else {
-                // Fallback
-                let estimated_padding = 24.0;
-                let text_height = 16.0;
-                estimated_padding + text_height
-            };
-
-            // Get spacing between header and list
-            let estimated_spacing = 4.0;
-
-            if total_items_in_list <= MAX_VISIBLE_ITEMS {
-                let single_item_height = if is_desktop { 32.0 } else { 64.0 };
-                let total_height =
-                    (total_items_in_list as f64 * single_item_height) + header_height + estimated_spacing;
-                popup.apply_over(cx, live! { height: (total_height) });
-            } else {
-                let max_height = if is_desktop { 400.0 } else { 480.0 };
-                popup.apply_over(cx, live! { height: (max_height) });
-            }
-
             // Only create widgets for items that will actually be visible
-            for (index, (display_name, member)) in matched_members.into_iter().take(MAX_VISIBLE_ITEMS).enumerate() {
+            // If @room exists, reserve one slot for it
+            let user_items_limit = if has_room_item { max_visible_items - 1 } else { max_visible_items };
```

Comment:
you can use the same `as usize` trick here. Also, I would use `saturating_sub` if it' possible for `max_visible_items` to ever be 0. But it's safer anyway.

```suggestion
            let user_items_limit = max_visible_items.saturating_sub(has_room_item as usize);
```